frontend/app/src/components/display/amount/AmountDisplay.vue

```diff
@@ -0,0 +1,404 @@
+<script setup lang="ts">
+import { BigNumber } from '@rotki/common';
+import { type ComputedRef, type Ref } from 'vue';
+import AmountCurrency from '@/components/display/AmountCurrency.vue';
+import { displayAmountFormatter } from '@/data/amount_formatter';
+import { CURRENCY_USD, type Currency, useCurrencies } from '@/types/currencies';
+import { One, Zero } from '@/utils/bignumbers';
+import { type RoundingMode } from '@/types/frontend-settings';
+
+const CurrencyType = ['none', 'ticker', 'symbol', 'name'] as const;
+type ShownCurrency = (typeof CurrencyType)[number];
+
+const props = withDefaults(
+  defineProps<{
+    value: BigNumber;
+    loading?: boolean;
+    amount?: BigNumber;
+    // This is what the fiat currency is `value` in. If it is null, it means we want to show it the way it is, no conversion, e.g. amount of an asset.
+    fiatCurrency?: string | null;
+    showCurrency?: ShownCurrency;
+    forceCurrency?: boolean;
+    asset?: string;
+    // This is price of `priceAsset`
+    priceOfAsset?: BigNumber | null;
+    // This is asset we want to calculate the value from, instead get it directly from `value`
+    priceAsset?: string;
+    integer?: boolean;
+    assetPadding?: number;
+    // This prop to give color to the text based on the value
+    pnl?: boolean;
+    noScramble?: boolean;
+    timestamp?: number;
+    /**
+     * Amount display text is really large
+     */
+    xl?: boolean;
+  }>(),
+  {
+    loading: false,
+    amount: () => One,
+    fiatCurrency: null,
+    showCurrency: 'none',
+    forceCurrency: false,
+    asset: '',
+    priceOfAsset: null,
+    priceAsset: '',
+    integer: false,
+    assetPadding: 0,
+    pnl: false,
+    noScramble: false,
+    timestamp: -1,
+    xl: false
+  }
+);
+
+const {
+  amount,
+  value,
+  asset,
+  fiatCurrency: sourceCurrency,
+  showCurrency,
+  priceOfAsset,
+  priceAsset,
+  integer,
+  forceCurrency,
+  noScramble,
+  timestamp
+} = toRefs(props);
+
+const evaluating: Ref<boolean> = ref(false);
+
+const {
+  currency,
+  currencySymbol: currentCurrency,
+  floatingPrecision
+} = storeToRefs(useGeneralSettingsStore());
+
+const { scrambleData, shouldShowAmount, scrambleMultiplier } = storeToRefs(
+  useSessionSettingsStore()
+);
+
+const { exchangeRate, assetPrice, getHistoricPrice } = useBalancePricesStore();
+
+const {
+  abbreviateNumber,
+  thousandSeparator,
+  decimalSeparator,
+  currencyLocation,
+  amountRoundingMode,
+  valueRoundingMode
+} = storeToRefs(useFrontendSettingsStore());
+
+const { isAssetPriceInCurrentCurrency } = useBalancePricesStore();
+
+const isCurrentCurrency = isAssetPriceInCurrentCurrency(priceAsset);
+
+const { findCurrency } = useCurrencies();
+
+const historicExchangeRate = asyncComputed(
+  async () => {
+    if (!isDefined(sourceCurrency)) {
+      return One.negated();
+    }
```

Comment:
`
    if (get(sourceCurrency) === get(currentCurrency)) {
      return One;
    }
`

I think it makes sense to put this here (?)
If the currency is the same, we don't need to call the historic price.

Reply:
Ok, the intent is not clearly communicated here.

The commented code is supposed to be a no-op for the compiler. I will turn it into an assertion instead, to that we don't change the signature of the `getHistoricPrice`.

Because the `historicPrice` `asyncComputed` is lazy it will only be evaluated when called. and when it is called the check has already occurred that the source and current currency are not the same.

Calling `getHistoricPrice` with the same asset here is a mistake it should not be called with the same from and to. We already do such checks before in the code.

I will turn both into assertions.